pkg/member/member.go

```diff
@@ -138,7 +138,7 @@ func (m *EmbeddedEtcdMember) GetLeader() *pdpb.Member {
 }
 
 // setLeader sets the member's PD leader.
-func (m *EmbeddedEtcdMember) setLeader(member *pdpb.Member) {
+func (m *EmbeddedEtcdMember) setLeader(member interface{}) {
```

Comment:
Will the generic type help to eliminate the empty interface?

Reply:
Restored the change here by adding abstractions ElectionLeader interface, EmbeddedEtcdLeader struct and EtcdLeader struct.

Reply:
BTW, to your original question, if you mean using generic type to consolidate EmbeddedEtcdMember and Participant, the answer mostly is no, because the former is tightly coupled with the embedded etcd, and the logic in some functions can't be reused.